packages/app/src/dimension-mapper/hooks.ts

```diff
@@ -1,10 +1,13 @@
-import { useState } from 'react';
+import { useDebouncedState } from '@react-hookz/web';
 
 import type { DimensionMapping } from './models';
 
 export function useDimMappingState(dims: number[], axesCount: number) {
-  return useState<DimensionMapping>([
-    ...Array.from({ length: dims.length - axesCount }, () => 0),
-    ...['y' as const, 'x' as const].slice(-axesCount),
-  ]);
+  return useDebouncedState<DimensionMapping>(
+    [
+      ...Array.from({ length: dims.length - axesCount }, () => 0),
+      ...['y' as const, 'x' as const].slice(-axesCount),
+    ],
+    50
```

Comment:
This number may be changed but `50` is already enough to prevent fetching when doing a continuous motion with the slider thumb